Mac/Linux examples for environment vars

masaok opened the pull request:
# Why

Adding examples for env vars

# How

I tested various ways of environment variables until it worked.  The current documentation is vague.

# Test Plan

No tests necessary.

# Checklist

- [x ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

masaok committed:
Mac/Linux examples for environment vars

brentvatne commented:
this probably deserves it's on section on "how to define environment variables on your system" that explains alternatives for how to do this on macos/linux/windows

ide committed:
Change env docs to use config values, not secrets

It is dangerous to include secret values in client apps (has nothing to do with Expo -- any values shipped to a client are, by definition, readable by the client).

ide commented:
Landed a change so that we do not suggest using secrets in environment variables. Any secrets shipped in a client app are compromised (has nothing to do with Expo -- any values shipped to a client are, by definition, readable by the client).

brentvatne commented:
we removed support for automatically populating these env vars in expo-cli a while ago: https://github.com/expo/expo-cli/blob/master/CHANGELOG.md#thu-jan-14-143405-2021-0100

> - [cli] Environment variables matching `EXPO_` or `REACT_NATIVE_` are no longer exposed publicly to the development-mode app or website ([#3063](https://github.com/expo/expo-cli/issues/3063))

i updated the docs to reflect that: https://github.com/expo/expo/commit/d4b427b04c8ef3a3f425e6514c7f73b78c8870a0
